File: 73. Set Matrix Zeroes/_73.py
class Solution:
    def setZeroes(self, matrix) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        row = set()
        col = set()
        for colindex, v in enumerate(matrix):
            for rowindex, value in enumerate(v):
                if value == 0:
                    row.add(rowindex)
                    col.add(colindex)

        for i in row:
            for j in range(len(matrix)):
                matrix[j][i] = 0

        for i in col:
            for j in range(len(matrix[0])):
                matrix[i][j] = 0


test = [[0,1,2,0],[3,4,5,2],[1,3,1,5]]
a = Solution()
a.setZeroes(test)
print(test)

File: 73. Set Matrix Zeroes/73.go
package leetcode

func setZeroes(matrix [][]int)  {
	row := make([]int, 0)
	col := make([]int, 0)

	for colindex,v := range matrix{
		for rowindex, value := range v{
			if value == 0{
				col = append(col, colindex)
				row = append(row, rowindex)
			}
		}
	}

	for _,v := range row{
		for i:=0; i<len(matrix); i++{
			matrix[i][v] = 0
		}
	}
	for _,v := range col{
		for i:=0; i<len(matrix[0]); i++{
			matrix[v][i] = 0
		}
	}
}